File: 2/b.ts
// const input = await Bun.file("test.txt").text();
const input = await Bun.file("input.txt").text();

let res = 0;

input.split("\n").forEach((line, i) => {
  const [first, second] = line.split(": ");
  const max = {
    red: 0,
    green: 0,
    blue: 0,
  } as Record<string, number>;
  second.split("; ").forEach((item) => {
    const rev = item.split(", ");
    console.log(rev);
    rev.forEach((item) => {
      const [num, color] = item.split(" ");
      max[color] = Math.max(max[color], parseInt(num));
    });
  });
  console.log(max);
  const power = max.red * max.green * max.blue;
  console.log(power);
  res += power;
});

console.log(res);

File: 2/b.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, _ := os.ReadFile("input.txt")

	res := 0

	for _, line := range strings.Split(string(file), "\n") {
		_, gamesText, _ := strings.Cut(line, ": ")
		cubeCount := map[string]int{
			"red":   0,
			"green": 0,
			"blue":  0,
		}
		for _, game := range strings.Split(gamesText, "; ") {
			for _, inst := range strings.Split(game, ", ") {
				numberStr, color, _ := strings.Cut(inst, " ")
				number, _ := strconv.Atoi(numberStr)
				if number > cubeCount[color] {
					cubeCount[color] = number
				}
			}
		}
		res += cubeCount["red"] * cubeCount["green"] * cubeCount["blue"]
	}
	fmt.Println(res)
}
